pytorch_lightning/trainer/supporters.py

```diff
@@ -514,14 +574,18 @@ def _nested_calc_num_data(data: Union[Mapping, Sequence], compute_func: Callable
     return compute_func(new_data)
 
 
-def prefetch_iterator(iterable: Iterable) -> Generator[Tuple[Any, bool], None, None]:
+def prefetch_iterator(iterable: Iterable,
+                      trainer: Optional['pl.Trainer'] = None) -> Generator[Tuple[Any, bool], None, None]:
     """
     Returns an iterator that pre-fetches and caches the next item.
     The values are passed through from the given iterable with an added boolean indicating if this is the last item.
     See `https://stackoverflow.com/a/1630350 <https://stackoverflow.com/a/1630350>`_
     """
     it = iter(iterable)
 
+    if trainer:
+        trainer.current_iterator = it
```

Comment:
can you explain what this workaround is about?
I can see that you are using it in the checkpoint connector but it is not clear to me what problem it is trying to solve.

Reply:
This can be removed ! This was used to not lose the Iterator reference as it contains the DataLoader Iterators.